lib/lfrfid/lfrfid_dict_file.c

```diff
@@ -0,0 +1,181 @@
+#include "lfrfid_dict_file.h"
+#include <storage/storage.h>
+#include <flipper_format/flipper_format.h>
+#include <lfrfid/tools/bit_lib.h>
+
+#define LFRFID_DICT_FILETYPE "Flipper RFID key"
+
+bool lfrfid_dict_file_save(ProtocolDict* dict, ProtocolId protocol, const char* filename) {
+    furi_check(protocol != PROTOCOL_NO);
+    Storage* storage = furi_record_open(RECORD_STORAGE);
+    FlipperFormat* file = flipper_format_file_alloc(storage);
+    size_t data_size = protocol_dict_get_data_size(dict, protocol);
+    uint8_t* data = malloc(data_size);
+    bool result = false;
+
+    do {
+        if(!flipper_format_file_open_always(file, filename)) break;
+        if(!flipper_format_write_header_cstr(file, LFRFID_DICT_FILETYPE, 1)) break;
+
+        // TODO: write comment about protocol types into file
+
+        if(!flipper_format_write_string_cstr(
+               file, "Key type", protocol_dict_get_name(dict, protocol)))
+            break;
+
+        // TODO: write comment about protocol sizes into file
+
+        protocol_dict_get_data(dict, protocol, data, data_size);
+
+        if(!flipper_format_write_hex(file, "Data", data, data_size)) break;
+        result = true;
+    } while(0);
```

Comment:
Minor but while(0) used here, while (false) used on 173, should likely be the same